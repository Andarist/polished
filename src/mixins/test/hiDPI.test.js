// @flow
import hiDPI from '../hiDPI'

describe('hiDPI', () => {
  beforeAll(() => {
    global.console = {
      error: jest.fn(),
      warn: jest.fn(),
      log: global.console.log,
    }
  })

  afterEach(() => {
    // eslint-disable-next-line no-console
    console.error.mockClear()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })

  it('should pass ratio to media query', () => {
    expect({
      [hiDPI(1.5)]: {
        width: '200px',
      },
    }).toMatchSnapshot()
  })

  it('should set a default ratio of 1.3 when no ratio is passed', () => {
    expect({
      [hiDPI()]: {
        width: '200px',
      },
    }).toMatchSnapshot()
  })

  it('should throw an error when called with a non-number value', () => {
    // $FlowIgnoreNextLine since the coming is invalid code, flow complains
    hiDPI('1px')
    // eslint-disable-next-line no-console
    expect(console.error.mock.calls).toMatchSnapshot()
  })

  it('should throw a warning when called with more than 1 parameter', () => {
    hiDPI(1, 2)
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
  })

  it('should throw an error and a warning when called with multiple string values', () => {
    // $FlowIgnoreNextLine since the coming is invalid code, flow complains
    hiDPI('1px', '2px')
    // eslint-disable-next-line no-console
    expect(console.error.mock.calls).toMatchSnapshot()
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
  })
})
