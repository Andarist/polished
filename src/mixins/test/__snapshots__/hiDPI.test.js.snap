exports[`hiDPI should pass ratio to media query 1`] = `
Object {
  "
    @media only screen and (-webkit-min-device-pixel-ratio: 1.5),
    only screen and (min--moz-device-pixel-ratio: 1.5),
    only screen and (-o-min-device-pixel-ratio: 1.5/1),
    only screen and (min-resolution: 144dpi),
    only screen and (min-resolution: 1.5dppx)
  ": Object {
    "width": "200px",
  },
}
`;

exports[`hiDPI should set a default ratio of 1.3 when no ratio is passed 1`] = `
Object {
  "
    @media only screen and (-webkit-min-device-pixel-ratio: 1.3),
    only screen and (min--moz-device-pixel-ratio: 1.3),
    only screen and (-o-min-device-pixel-ratio: 1.3/1),
    only screen and (min-resolution: 125dpi),
    only screen and (min-resolution: 1.3dppx)
  ": Object {
    "width": "200px",
  },
}
`;

exports[`hiDPI should throw a warning when called with more than 1 parameter 1`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- mixins/hiDPI.js --%c

hiDPI %cexpects a maximum of %c1 parameter%c. However, you passed %c2 parameters%c. These additional parameters were ignored.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#hiDPI%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: green",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: goldenrod",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;

exports[`hiDPI should throw an error and a warning when called with multiple string values 1`] = `
Array [
  Array [
    "%c ✨ ERROR ✨ ---- mixins/hiDPI.js --%c

hiDPI %cexpects a parameter of type number. However, you passed 1px(string) instead.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#hiDPI%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;

exports[`hiDPI should throw an error and a warning when called with multiple string values 2`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- mixins/hiDPI.js --%c

hiDPI %cexpects a maximum of %c1 parameter%c. However, you passed %c2 parameters%c. These additional parameters were ignored.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#hiDPI%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: green",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold; color: goldenrod",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;

exports[`hiDPI should throw an error when called with a non-number value 1`] = `
Array [
  Array [
    "%c ✨ ERROR ✨ ---- mixins/hiDPI.js --%c

hiDPI %cexpects a parameter of type number. However, you passed 1px(string) instead.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#hiDPI%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;
